README.md

```diff
@@ -37,9 +35,36 @@ multiple Terraform modules, remote state, and locking.
    Terraform you already have installed. However, based on the settings in your `terraform.tfvars` file, Terragrunt can
    configure remote state, locking, extra arguments, and lots more.
 
+1. For some optional, yet useful, background reading that will help explain the motivation for Terragrunt, see: [How to create reusable infrastructure with Terraform modules](https://blog.gruntwork.io/how-to-create-reusable-infrastructure-with-terraform-modules-25526d65f73d) and [How to use Terraform as a team](https://blog.gruntwork.io/how-to-use-terraform-as-a-team-251bc1104973).
+
+## Table of Contents
+
+1. [Install Terragrunt](#install-terragrunt)
+1. [Use cases](#use-cases)
+   1. [Keep your Terraform code DRY](#keep-your-terraform-code-dry)
+   1. [Keep your remote state configuration DRY](#keep-your-remote-state-configuration-dry)
+   1. [Keep your CLI flags DRY](#keep-your-cli-flags-dry)
+   1. [Work with multiple Terraform modules](#work-with-multiple-terraform-modules)
+1. [Terragrunt details](#terragrunt-details)
+   1. [Interpolation Syntax](#interpolation-syntax)
+   1. [CLI options](#cli-options)
+   1. [Configuration](#configuration)
+   1. [Developing Terragrunt](#developing-terragrunt)
+   1. [License](#license)
+
```

Comment:
Yea, agreed that keeping it updated will be a bit of a PITA. We can look into some tools to automate this, perhaps in a pre-commit hook?

Seems like there is a lot of conversation about support of ToC over in https://github.com/isaacs/github/issues/215 and https://github.com/github/markup/issues/904. Several tools mentioned, which we could try out.